atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/bucketingthings/DefaultSweepAssignedBucketStore.java

```diff
@@ -174,6 +185,32 @@ public void updateStartingBucketForShardAndStrategy(Bucket newStartingBucket) {
         }
     }
 
+    @Override
+    public Set<SweepableBucket> getSweepableBuckets(Set<Bucket> startBuckets) {
+        List<SweepAssignedBucketsRow> rows = startBuckets.stream()
+                .flatMap(bucket -> Stream.of(
+                        SweepAssignedBucketStoreKeyPersister.INSTANCE.sweepBucketsRow(bucket),
+                        SweepAssignedBucketStoreKeyPersister.INSTANCE.nextSweepBucketsRow(bucket)))
+                .collect(Collectors.toList());
+
+        return readSweepableBucketRows(rows);
+    }
+
+    @Override
+    public void putTimestampRangeForBucket(
+            Bucket bucket, Optional<TimestampRange> oldTimestampRange, TimestampRange newTimestampRange) {
+        Cell cell = SweepAssignedBucketStoreKeyPersister.INSTANCE.sweepBucketsCell(bucket);
+        casCell(
+                cell,
+                oldTimestampRange.map(timestampRangePersister::trySerialize),
+                timestampRangePersister.trySerialize(newTimestampRange));
+    }
+
+    @Override
+    public void deleteBucketEntry(Bucket bucket) {
+        throw new UnsupportedOperationException("deleteBucketEntry is not implemented yet.");
```

Comment:
question: we need #7197 or similar for this right?

Reply:
Yes